hack/v1migration/delete.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+WORKDIR=${1:?must specify working directory}
+echo "Using workdir: $WORKDIR"
+
+case "${2}" in
+"--dry-run" )
+  dry_run="--dry-run"
+  ;;
+"" )
+  # When doing a dry run, allow the script to continue after errors. Operations such as deleting a CRD are likely
+  # to fail since the CRs are not being deleted in a dry run.
+  set -e
+  ;;
+* )
+  echo "Only valid flag is \"--dry-run\""
+  exit 1
+esac
+
+if [[ -z $dry_run ]]
+then
+  source "$(dirname $0)/verify_scaledown.sh"
+  verify_all_scaled_down
+fi
+
+mkdir -p "$WORKDIR"
+
+if [[ -z $dry_run && \
+      "$(ls "$WORKDIR" 2>/dev/null)" ]]
+then
+  echo "WARNING: workdir ($WORKDIR) is not empty."
+  echo "Files in workdir may be overwritten."
+  read -r -p "Do you wish to proceed? [y/N] " response
+  if [[ ! "$response" =~ ^([yY]([eE][sS])?)$ ]]
+  then
+    exit 1
+  fi
+fi
+
+readarray -t HIVE_TYPES <<< \
+  "$(oc get crd -o json | \
+    jq -r '.items[].spec | select((.group="hive.openshift.io") and (.version="v1alpha1")) | .names.plural')"
+
+for t in "${HIVE_TYPES[@]}"
+do
+	echo "Storing all ${t} in ${WORKDIR}/${t}.json"
+	oc get "${t}.hive.openshift.io" -A -o json | jq .items[] > "${WORKDIR}/${t}.json"
+done
+
+bin/hiveutil v1migration save-owner-refs --work-dir "${WORKDIR}"
+
+if [[ -z $dry_run ]]
+then
+  echo "WARNING: this script will delete all Hive custom resources and their definitions"
+  echo "It should only be used during migrations to the v1 API."
+  read -r -p "Do you wish to proceed? [y/N] " response
+  if [[ ! "$response" =~ ^([yY]([eE][sS])?)$ ]]
+  then
+    exit 1
+  fi
+fi
+
+if [[ -z $dry_run ]]
+then
+  echo "Deleting install jobs"
+  oc delete jobs -l "hive.openshift.io/install=true" -A
```

Comment:
If one of these was in progress when we killed it, when we restore the controller should see the missing job for a clusterprovision, and recreate it I assume. InfraID for in-progress looks like it's still stores in ClusterProvision.Spec.InfraID so should be restored, and cleaned up.

Reply:
The clusterprovision controller will mark the ClusterProvision as failed due to a lost job. The clusterdeployment controller will then create a new ClusterProvision. If the provision had made it past the initializing phase and starting creating resources in the cloud, then the infra ID would have to have been set on the ClusterDeployment. So the next ClusterProvision will get a prevInfraID and clean up the resources from the killed provision attempt.